arch/arm/core/aarch64/cache.S

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * ARM64 Cortex-A Cache management
+ */
+
+#include <toolchain.h>
+#include <linker/sections.h>
+#include <arch/cpu.h>
+
+_ASM_FILE_PROLOGUE
+
+.macro	dcache_range_op, dcache op
+	/* Calculate dcache line size */
+	mrs	x3, ctr_el0
```

Comment:
Isn't this redundant given we have `arch_dcache_line_size_get` already?

Reply:
Linux Kernel use similar method, embed cache line calculation in cache functions. arch_dcache_line_size_get is for c code usage.